core/src/emr-eks-platform/resources/k8s/karpenter-provisioner-config/critical-provisioner.yml

```diff
@@ -0,0 +1,69 @@
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: critical-{{az}}
+spec:
+  # References cloud provider-specific custom resource, see your cloud provider specific documentation
+  providerRef:
+    name: critical-nodes-{{az}}
+
+  # Labels are arbitrary key-values that are applied to all nodes
+  labels:
+    role: critical
+    node-lifecycle: on-demand
+  
+  taints:
+    - key: role
+      value: critical
+      effect: NoSchedule
+
+  # Requirements that constrain the parameters of provisioned nodes.
+  # These requirements are combined with pod.spec.affinity.nodeAffinity rules.
+  # Operators { In, NotIn } are supported to enable including or excluding values
+  requirements:
```

Comment:
this is different from the manage nodegroups, ca we make it consistent? The biggest issue is on m or r instance type which change the spark conf to maximize the resources so it could lead to waste of resources because executors are homogeneous